src/main/java/com/google/plantasy/servlets/QuizPoints.java

```diff
@@ -37,17 +37,18 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
             response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
             return;
         }
-
-        //Adds points to the user who took the quiz
-        Long quiz_time = QuizTimingPropertiesUtils.getQuizTimestampProperty("Game", "gameId", userEntity.getProperty("gameId").toString(), datastore);
-        Long user_time = QuizTimingPropertiesUtils.getQuizTimestampProperty("user", "userID", userEntity.getProperty("userID").toString(), datastore);
-        QuizTimingPropertiesUtils.giveUserQuizTakenPoints(QuizTimingPropertiesUtils.userTookQuiz(user_time, quiz_time), userEntity, datastore);        
-
+ 
+        //Adds points to the user who took the quiz & update their timestamp
+        User current_user = new User(userEntity);
+        UserUtils.addPoints(current_user, 20, datastore);
+        current_user.setQuizTiming(System.currentTimeMillis());
+        datastore.put(current_user.getEntity());
+        
         //Adds points to the user who got voted for in the quiz 
         String clicked_user_id = HttpRequestUtils.getParameterWithDefault(request, "user_picture", "");
         Entity user_clicked = UserUtils.getEntityFromDatastore("user", "userID", clicked_user_id, datastore);
-        User user = new User(user_clicked);
-        UserUtils.addPoints(user, 20, datastore);
+        User user_click = new User(user_clicked); 
+        UserUtils.addPoints(user_click, 20, datastore);
```

Comment:
nit: should be user_clicked or selected_user or something like that.

Reply:
Changed this!